functions/api/action.js

```diff
@@ -0,0 +1,208 @@
+const constants = require('../common/constants')
+const validate = require('../common/validator')
+const logger = require('../middleware/logger');
+const joi = require('@hapi/joi');
+const admin = require('firebase-admin');
+const functions = require('firebase-functions');
+const express = require('express');
+const { data } = require('../middleware/logger');
+const router = express.Router();
+const db = admin.firestore();
+
+/**
+ * Route to perform addProduct transaction
+ * @returns 201 Created
+ */
+router.post('/addProduct', async (request, response, next) => {
+    logger.info(`Adding product to inventory....`);
+    // Validate parameters
+    logger.debug('Validating params.')
+    const { error } = validateParams(request.body)
+    if (error) {
+        const err = new Error(error.details[0].message)
+        err.statusCode = 400
+        next(err)
+        return;
+    }
+
+    const data = request.body
+    data[constants.OPERATION] = constants.ADD_PRODUCT
+    data[constants.USER] = request.user.email
+    data[constants.DATE] = new Date()
+
+    const transactionId = await module.exports.createTransaction(data)
```

Comment:
general qz with all action endpoints. Based on request body , you are creating transaction. Are you updating the inventory as well?

Reply:
Yes. Inventory is updated using a trigger. It updates availableQuantity and isBelowThreshold